jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/applications/AbstractVariantEndpoint.java

```diff
@@ -49,35 +43,46 @@
     @Inject
     protected GenericVariantService variantService;
 
+    // required for RESTEasy
+    public AbstractVariantEndpoint() {
+    }
+
+    // required for tests
+    protected AbstractVariantEndpoint(Validator validator, SearchManager searchManager) {
+        super(validator, searchManager);
+    }
+
     /**
      * Secret Reset
      *
      * @param variantId id of {@link Variant}
      * @return          {@link Variant} with new secret
      *
      * @statuscode 200 The secret of Variant reset successfully
+     * @statuscode 400 The requested Variant resource exists but it is not of the given type
      * @statuscode 404 The requested Variant resource does not exist
      */
-    @PUT
-    @Path("/{variantId}/reset")
-    @Consumes(MediaType.APPLICATION_JSON)
-    @Produces(MediaType.APPLICATION_JSON)
-    public Response resetSecret(@PathParam("variantId") String variantId) {
+    protected <T extends Variant> Response doResetSecret(String variantId, Class<T> type) {
```

Comment:
changing the constructor as suggested before, this can be changed again to 
```java
public Response resetSecret(@PathParam("variantId") String variantId)
```